### ISSUE TITLE
Crash in ensureCachedScrollSelectorValue

### ISSUE THREAD

**Issue by fabads** [ShawnLin013/NumberPicker#100]
Hi ShawnLin,

I tried to test your proposal for Issue #99 and found another problem when trying to update the min/max/displayedvalues. Here the test case:

```
mFreezones = new ArrayList<>(Arrays.asList("01", "02", "03", "04","05","06","07","08","09","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24"));
mFreezones.remove("01");
mFreezones.remove("02");
mFreezones.remove("03");
mFreezones.remove("04");
mFreezones.remove("05");

mZoneNumberPicker.setMinValue(1);
mZoneNumberPicker.setMaxValue(mFreezones.size());
mZoneNumberPicker.setDisplayedValues(mFreezones.toArray(new String[0]));
mZoneNumberPicker.setValue(1);

mFreezones = new ArrayList<>(Arrays.asList("01", "02", "03", "04","05","06","07","08","09","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24"));
mFreezones.remove("01");
mFreezones.remove("02");
mFreezones.remove("03");
mFreezones.remove("04");

mZoneNumberPicker.setMinValue(1);
mZoneNumberPicker.setMaxValue(mFreezones.size());
mZoneNumberPicker.setDisplayedValues(mFreezones.toArray(new String[0]));
mZoneNumberPicker.setValue(1);
```

The crash happens when calling the second setMaxValue function.
In my real code, the number picker is let's say a zone number picker. 
I need to update the zone number picker values each time my database is changed (let's say another user delete a "zone", this zone can now be used by my user, so need to update the freezones displayed by the zone number picker).

Maybe I'm wrong in the ways to use your number picker, let me know!

**Comment by UnateD** [comment]
Hello, try wrap setMaxValue call in try{} catch{Throwable e} when you set new values . Bad move, but it works.

**Comment by ShawnLin013** [comment]
Fixed at v2.4.9.
